usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedStopAll.java

```diff
@@ -0,0 +1,317 @@
+     * stop entities is not respected.
+     *
+     * @param parsedEntityByStopId a map of entities to be processed
+     */
+    public void execute(final Map<String, ParsedEntity> parsedEntityByStopId) {
+        Map<String, List<String>> childrenPerStationId = new HashMap<>();
+
+        // first we iterate through all to resolve children
+        parsedEntityByStopId.values().forEach(stop -> {
+            final String childId = stop.getEntityId();
+            final String parentId = (String) stop.get("parent_station");
+            if (parentId != null) {
+                if (childrenPerStationId.containsKey(parentId)) {
+                    childrenPerStationId.get(parentId).add(childId);
+                } else {
+                    List<String> childrenList = new ArrayList<>();
+                    childrenList.add(childId);
+                    childrenPerStationId.put(parentId, childrenList);
+                }
+            }
+        });
+
+        // then we transform rows to the corresponding concrete types, base on 'location_type'
+        parsedEntityByStopId.values().forEach(stop -> {
+            Integer locationType = (Integer) stop.get("location_type");
+            String stopId = stop.getEntityId();
+            String stopName = (String) stop.get("stop_name");
+            Float stopLat = (Float) stop.get("stop_lat");
+            Float stopLon = (Float) stop.get("stop_lon");
+            String stopCode = (String) stop.get("stop_code");
+            String stopDesc = (String) stop.get("stop_desc");
+            String zoneId = (String) stop.get("zone_id");
+            String stopUrl = (String) stop.get("stop_url");
+            String parentStation = (String) stop.get("parent_station");
+            String stopTimezone = (String) stop.get("stop_timezone");
+            Integer wheelchairBoarding = (Integer) stop.get("wheelchair_boarding");
+            String levelId = (String) stop.get("level_id");
+            String platformCode = (String) stop.get("platform_code");
+
+            EntityBuildResult<?> stopEntityBuildResult = null;
+
+            switch (LocationType.fromInt(locationType)) {
+                case STOP_OR_PLATFORM: {
+                    if (parentStation != null) {
+                        ParsedEntity parent = parsedEntityByStopId.get(parentStation);
+                        Integer parentLocationType = (Integer) parent.get("location_type");
+
+                        if (LocationType.fromInt(parentLocationType) != LocationType.STATION) {
+                            resultRepository.addNotice(
+                                    new ParentStationInvalidLocationTypeNotice(
+                                            stopId, locationType, parentStation, 1,
+                                            parentLocationType
+                                    )
+                            );
+                        }
+
+                        if (WheelchairBoarding.fromInt(wheelchairBoarding) ==
```

Comment:
Stops also inherit parent `stop_timezone` - from https://gtfs.org/reference/static#stopstxt:

>If the location has a parent station, it inherits the parent station’s timezone instead of applying its own. Stations and parentless stops with empty stop_timezone inherit the timezone specified by agency.agency_timezone. If stop_timezone values are provided, the times in stop_times.txt should be entered as the time since midnight in the timezone specified by agency.agency_timezone.

Reply:
Good catch! Some additional work is required. I will not merge this PR then

Reply:
Do you know what's the expected behavior if there are multiple agencies in agency.txt?

Reply:
Ok I got my answer from the spec

> If multiple agencies are specified in the dataset, each must have the same agency_timezone.

Reply:
Implemented in last commit
- location_type=0 -> inherit from parent if there is one. Otherwise inherit from agency if stop_timezone is not provided
- location_type==1 -> inherit from agency if stop_timezone is not provided
- location_type==2 -> inherit from parent if there is one
- location_type==3 -> inherit from parent if there is one
- location_type==4 -> inherit from parent if there is one